design/src/docs/anatomy.md

```diff
@@ -0,0 +1,601 @@
+
+        Op4: Upgradable<AwsRestJson1, GetServerStatistics>,
+        Op4::Service: tower::Service<http::Request, Error = Infallible>,
+
+        Op5: Upgradable<AwsRestJson1, CapturePokemon>,
+        Op5::Service: tower::Service<http::Request, Error = Infallible>,
+
+        Op6: Upgradable<AwsRestJson1, GetStorage>,
+        Op6::Service: tower::Service<http::Request, Error = Infallible>,
+```
+
+Notice the general form: `Op{N}` must upgrade to a HTTP service.
+
+We provide two builder constructors:
+
+```rust
+    /// Constructs a builder for [`PokemonService`].
+    pub fn builder() -> PokemonServiceBuilder<
+        MissingOperation,
+        MissingOperation,
+        MissingOperation,
+        MissingOperation,
+        MissingOperation,
+        MissingOperation,
+    > {
+        PokemonServiceBuilder {
+            check_health: MissingOperation,
+            do_nothing: MissingOperation,
+            get_pokemon_species: MissingOperation,
+            get_server_statistics: MissingOperation,
+            capture_pokemon: MissingOperation,
+            get_storage: MissingOperation,
+        }
+    }
+
+    /// Constructs an unchecked builder for [`PokemonService`].
+    ///
+    /// This will not enforce that all operations are set, however if an unset operation is used at runtime
+    /// it will return status code 500 and log an error.
+    pub fn unchecked_builder() -> PokemonServiceBuilder<
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+        FailOnMissingOperation,
+    > {
+        PokemonServiceBuilder {
+            check_health: FailOnMissingOperation,
+            do_nothing: FailOnMissingOperation,
+            get_pokemon_species: FailOnMissingOperation,
+            get_server_statistics: FailOnMissingOperation,
+            capture_pokemon: FailOnMissingOperation,
+            get_storage: FailOnMissingOperation,
+        }
+    }
+```
+
+The `builder` constructor provides a `PokemonServiceBuilder` where `build` cannot be called until all operations are set because `MissingOperation` purposefully doesn't implement `Upgradable`. In contrast, the `unchecked_builder` which sets all `Op{N}` to `FailOnMissingOperation` can be immediately built, however any unset operations are upgraded into a service which always returns status code 500, as noted in [Upgrading a Model Service](#upgrading-a-model-service).
+
+After all `Op{N}` are upgraded in `build` they are collected into their protocol specific `Router` implementation and then bundled up into a `RoutingService`. The `RoutingService` is then wrapped in a `PokemonService` newtype and presented to the user.
+
+```rust
+/// The Pokémon Service allows you to retrieve information about Pokémon species.
+#[derive(Clone)]
+pub struct PokemonService<S> {
+    router: RoutingService<RestRouter<S>, AwsRestJson1>,
+}
+```
+
+## Plugins
```

Comment:
There should be a standalone tutorial on actually writing a `Plugin`. And when there is we should link to it here.

Add a TODO.